### PR TITLE
🔥 ts parse

### PR DESCRIPTION
**Pull request by robrix**
This PR:

- [x] 🔥s the `semantic ts-parse` subcommand. We only ever used it for debugging `Assignment`, which we’re currently in the process of removing.
- [x] 🔥s `Semantic.AST`.
- [x] 🔥s the various `*ASTParser` bindings in `Parsing.Parser`.
- [x] Depends on https://github.com/tree-sitter/haskell-tree-sitter/pull/235.
- [x] Depends on a release of `tree-sitter` 0.5.
- [x] Depends on a release of `tree-sitter-java` 0.4.
- [x] Depends on a release of `tree-sitter-json` 0.3.
- [x] Depends on a release of `tree-sitter-python` 0.6.
- [x] Depends on #313.

### CODE REVIEW

File: src/Parsing/Parser.hs [hunk 1]
```diff
@@ -112,43 +106,31 @@ data Parser term where
   UnmarshalParser :: Unmarshal t => Ptr TS.Language -> Parser (t Loc)
   -- | A parser producing an à la carte term given an 'AST'-producing parser and an 'Assignment' onto 'Term's in some syntax type.
   AssignmentParser :: (Enum grammar, Ix grammar, Show grammar, TS.Symbol grammar, Syntax.Error :< fs, Eq1 ast, Apply Foldable fs, Apply Functor fs, Foldable ast, Functor ast)
-                   => Parser (Term ast (Node grammar))           -- ^ A parser producing AST.
+                   => Parser (AST ast grammar)                   -- ^ A parser producing AST.
                    -> Assignment ast grammar (Term (Sum fs) Loc) -- ^ An assignment from AST onto 'Term's.
                    -> Parser (Term (Sum fs) Loc)                 -- ^ A parser producing 'Term's.
   -- | A parser for 'Markdown' using cmark.
-  MarkdownParser :: Parser (Term (TermF [] CMarkGFM.NodeType) (Node Markdown.Grammar))
+  MarkdownParser :: Parser (AST (TermF [] CMarkGFM.NodeType) Markdown.Grammar)
```
Comment (robrix) [inline src/Parsing/Parser.hs]: Using the same synonym consistently makes it a little clearer that `ASTParser` and `MarkdownParser` plug into `AssignmentParser`.

File: src/Semantic/CLI.hs [hunk 2]
```diff
@@ -149,25 +148,6 @@ parseCommand = command "parse" (info parseArgumentsParser (progDesc "Generate pa
...
-      filesOrStdin <- FilesFromGitRepo
-                      <$> option str (long "gitDir" <> help "A .git directory to read from")
-                      <*> option shaReader (long "sha" <> help "The commit SHA1 to read from")
-                      <*> ( ExcludePaths <$> many (option str (long "exclude" <> short 'x' <> help "Paths to exclude"))
-                        <|> ExcludeFromHandle <$> flag' stdin (long "exclude-stdin" <> help "Exclude paths given to stdin")
-                        <|> IncludePaths <$> many (option str (long "only" <> help "Only include the specified paths"))
-                        <|> IncludePathsFromHandle <$> flag' stdin (long "only-stdin" <> help "Include only the paths given to stdin"))
-                  <|> FilesFromPaths <$> some (argument filePathReader (metavar "FILES..."))
-                  <|> pure (FilesFromHandle stdin)
-      pure $ Task.readBlobs filesOrStdin >>= AST.runASTParse format
```
Comment (robrix) [inline src/Semantic/CLI.hs]: 🔥

**Comment by robrix** [comment]
Oh… I didn’t see that #319 was based on this. So now the PR includes both. Whoops.

**Comment by patrickt** [comment]
@robrix My bad! I should have put a checkbox next to #319.

**Pull request merged**